jobs/build.Jenkinsfile

```diff
@@ -178,24 +178,28 @@ lock(resource: "build-${params.STREAM}") {
             """)
         }
 
+        // Determine parent version/commit information
         def parent_version = ""
         def parent_commit = ""
-        stage('Fetch') {
-            if (s3_stream_dir) {
-                pipeutils.aws_s3_cp_allow_noent("s3://${s3_stream_dir}/releases.json", "tmp/releases.json")
-                if (utils.pathExists("tmp/releases.json")) {
-                    def releases = readJSON file: "tmp/releases.json"
-                    // check if there's a previous release we should use as parent
-                    for (release in releases["releases"].reverse()) {
-                        def commit_obj = release["commits"].find{ commit -> commit["architecture"] == basearch }
-                        if (commit_obj != null) {
-                            parent_commit = commit_obj["checksum"]
-                            parent_version = release["version"]
-                            break
-                        }
+        if (s3_stream_dir) {
+            pipeutils.aws_s3_cp_allow_noent("s3://${s3_stream_dir}/releases.json", "tmp/releases.json")
+            if (utils.pathExists("tmp/releases.json")) {
+                def releases = readJSON file: "tmp/releases.json"
+                // check if there's a previous release we should use as parent
+                for (release in releases["releases"].reverse()) {
+                    def commit_obj = release["commits"].find{ commit -> commit["architecture"] == basearch }
+                    if (commit_obj != null) {
+                        parent_commit = commit_obj["checksum"]
+                        parent_version = release["version"]
+                        break
                     }
                 }
+            }
+        }
 
+        // buildfetch previous build info
+        stage('BuildFetch') {
```

Comment:
Bikeshed: "BuildFetch" looks weird... maybe just "Fetch Metadata" ?

Reply:
Because I'm intimately familiar with the code here "BuildFetch" is more meaningful to me because it runs a `cosa buildfetch` and I know what that does.

I agree that to untrained eyes "Fetch Metadata" might be more obvious.

I'm ok changing it if you want to in a followup.